core/server/services/auth/session/middleware.js

```diff
@@ -20,11 +20,11 @@ const getOrigin = (req) => {
         return origin;
     }
 
-    try {
```

Comment:
Why was this try/catch block removed? Seems like `url.parse()` still [throws](https://github.com/nodejs/node/blob/8c70b2084ce5f76ea1e3b3c4ccdeee4483fe338b/lib/url.js#L150) or does `req.get()` return an empty string when the header is not set (sorry don't remember from to of my head) :thinking: ?

Reply:
The try/catch block before was because `URL` throws if the thing passed is not a valid url - here we only need a string! We add a check for `protocol` and `host` to ensure that the string we passed (and we're about to return) was/is a valid url.

If the referrer header is not present then we return null on line 16.